### PR TITLE
Revision of guidelines

### PR DESCRIPTION
**Pull request by qreeves**
Recent events has caused a discussion on the clarity and boundaries of the guidelines for which it has been suggested that a few revisions be made to it. This is just what I have so far and is in no way officially representative of what the final document will be, it is open for community input in order to create a more robust set of guidelines that everyone feels comfortable with.

### CODE REVIEW

**Comment by swooboo** [comment]
I like this one. It does introduce more clarity.

**Comment by JeffBobbo** [comment]
https://github.com/red-eclipse/base/pull/703/files#diff-2e07dd34b69681fe15193412d1cb6639R98
> Note that arguing with moderators or being
> otherwise antagonistic via in-game communication is not allowed.
I'd specify about arguing about moderation-related decisions. This rule otherwise infers that I can't argue with a moderator about anything, including the best way to do eggs.

Likewise further down when arguing with players.

I'd also include quarantining players in the list of possible punishments.

**Comment by TheAssassin** [comment]
> Note that arguing with moderators or being otherwise antagonistic via in-game communication is not allowed.

I agree that this is too restrictive, and suggest another wording:

> Please note that arguing with moderators is pointless, and continuing to do so or being otherwise antagonistic via in-game communication after a warning is not allowed and might lead to moderating measures.

----

> Note that arguing or being otherwise antagonistic with players is not advised.

I also suggest another wording for this:

> Moderators should avoid arguing or being otherwise antagonistic with players. As they represent the game, they should take responsibility and try their best to provide a positive and welcoming atmosphere.

-----

Another point I'd like to see covered is that moderators should contact each other internally before undoing any measure, and only if that is impossible within a reasonable amount of time (should be more specific on that, suggestions welcome), undo and notify the other moderator about that, too. Otherwise, the authority of not just the moderator but the whole Red Eclipse team suffers a lot.

----

>  * If the gameplay of the server is modified substantially this must be indicated clearly, preferably through the use of "servermotd".

I'd like to `s/servermotd/serverdesc/`, or insert `serverdesc` before that. It's always better to allow a player to see it at a first glance usability wise.

----

>  * The server must not advertise or otherwise spam messages excessively.

I'd like to insert this amendment:

> Announcements for instance, such as about upcoming events, are allowed as long as they aren't sent too often so that they could be interpreted as spam.

We should consider narrowing the word "excessively" to some actual value.

----

I'd also like to add a paragraph about being able to quarantine players if they ruin a team's experience, if they are explained the reasons and seem to be listening to the instructions and tips of the moderators. Currently this is definitely covered in some way by the paragraph about *disruptive behavior*, so I think it should be a subitem. An explicit rule about that will help a lot.

-----

As we're at it at the moment, we should implement the suggestions made in the forums, too. https://redeclipse.net/forum/viewtopic.php?f=4&t=1120

**Comment by zaquest** [comment]
> I'd also like to add a paragraph about being able to quarantine players if they ruin a team's experience

This is too broad and ambiguous, unless a list of specific actions will be given. Like team killing. Since we're currently having issues with chat killing, one such action might be: excessive chatting while alive that affects team's performance (like when one is chat killed in survivor, or chat killed with a flag/bomb).

**Comment by acerspyro** [comment]
@zaquest The kind of chat killing you're talking about should be gobbled up under the same category as "Being purposefully unhelpful to your team". I do see from time to time players going in circles instead of playing the actual game, actually.

**Comment by zaquest** [comment]
> Being purposefully unhelpful to your team

This is still too vague as it leaves open the question about what is considered "purposeful". Maybe the guy in your case tried to learn the map and improve his parkour, or maybe he failed to understand the objective. People have different abilities, hence a list of *specific actions* that are forbidden should be given.

**Comment by TheAssassin** [comment]
@zaquest the point is, I often try to instruct people, and tell them, and 90% follow them and get better. There are these days though, when some player *all the time* brings the flag to the wrong base and stuff like "ok, don't pick it up again" don't work. This is why I want the `/warn` command, too, and I'm waiting for @qreeves to implement that in NNUI (he promised to do it). We don't need that rule, we've got it. We should just explicitly mention it in my opinion.

As you have not explicitly disagreed but asked for another wording: Go get active and suggest something, that'll probably work better than criticism about specific passages.

**Comment by acerspyro** [comment]
On the topic of chat killing, we should simply not ban it, but instead, let the player know that it's wrong with a little sound or notification when you shoot a player while they're chatting, the same sound and effects we get by shooting a team mate.

This idea is pretty much stolen from Xonotic, where shooting a chatter causes two things:
- Error sound gets played (Xonotic's bots also yell "Check your colors!" or whatever :P )
- The death message changes when a player has been chat killed (Player1 was typekilled by Player2!)

But the game does not actually act upon chat killers. Simply letting them know that it's morally wrong to chat kill is all that it takes, really. The newbies don't know, and cannot be expected to know that.

**Comment by TheAssassin** [comment]
Then we need to describe that in the guidelines. Otherwise, some players might still try to argue against it "for not being forbidden" (seriously, the guidelines are nothing that grants any rights, it rather creates duties for both parties).

But well, I usually reply, "hey, stop shooting and moving, it's not allowed by the guidelines".

**Comment by zaquest** [comment]
@TheAssassin I don't actually think this should be moderated beyond team killing. So I can't possibly know what exactly you're trying to put under that flag of "ruining team's experience". So if you could provide a list of actions that you think are unacceptable I could help with reducing ambiguity in it. And I have already suggested an example of an action that fits in ruining team's experience.

**Comment by JeffBobbo** [comment]
>Then we need to describe that in the guidelines. Otherwise, some players might still try to argue against it "for not being forbidden" (seriously, the guidelines are nothing that grants any rights, it rather creates duties for both parties).

No, it creates a scenario where I can hit 'T' and be like "Hey, that's against the rules!", report it on the forums and get the player banned.

**Comment by TheAssassin** [comment]
Not really, as I am not talking about forbidding it, but people *pretending* that it was forbidden. Re-read my statement carefully please. As these people do not have any permissions, their claims can be discarded anyway.

I intended to suggest a wording like "Chatkilling is generally discouraged, and should be avoided except for rare situations. Especially when the chatting person is not a threat to oneself or one's team (such as when in-game support is provided), chatkills are a bad way to react. Everyone needs help from time to time."

The whole "go to spectator" discussion is (pardon my wording) bullshit. Since you wanted the stats to be true, then, they must not have an effect on the deaths, and the spectated times must be not be counted by the stats. This is over-engineering par excellence. **There is no reason to do so, I just explained why, so please don't misinterpret my words as a suggestion.** I do have some belief in human mankind (even on the Internet), and most people will respect the rule. We have just found a single case in which this did not work, but this person might be convinced that it has something to do with respect. The new "respect the moderators" wording (which I think should have and actually has been done in the past anyway) matches these cases, so that actions can be taken in extreme cases.

**Comment by acerspyro** [comment]
I find the concept of _Respect_ to be very ambiguous across people and cultures. I'll be surprised if someone manages to efficiently target the concept of _Disrespect_ in the rules' legalese. Best thing to do if you guys REALLY want to bust out disrespectful people would be to hold a consensus between administrators deciding what to do of a person considered to be disrespectful.

This, once again, overcomplicates everything in my opinion. I say we follow what I said earlier, and let nature follow its due course for the rest of the consequences.

**Comment by TheAssassin** [comment]
I don't like having to say this, but we need that respect rule unfortunately. Otherwise, personal attacks and these shitposts claiming about abuse which depict the situation totally wrong (reductio ad absurdum sends their regards) will continue. Most people don't get muted for in-game actions (that's totally absurd), they get muted for being disrespectful or for personal attacks, or for trying to influence others in a bad way.

**Comment by acerspyro** [comment]
@TheAssassin Personal attacks are easy to define in legalese. Defining being disrespectful is a bit harder. What do you mean by "Trying to influence others in a bad way"?

**Comment by TheAssassin** [comment]
An attempt to define some aspects of "toxic behavior". Some players try to manipulate others, and some with a bad attitude post their claims, which are not true (a lie) or can be misinterpreted easily with intention, and players might believe that stuff (people tend to believe the "fact" that sounds worst (i.e. "alternative facts"). That ruins the community atmosphere. We have had people in the ban and mute list, and I think currently have an entry because of this. We need such a rule to be able to remove toxic "elements" from the game.

I want to cite @JeffBobbo here:
> acerspyro He means players influencing bad behaviour in other players.
> -- Red Eclipse chat

Side note: It's sad that some moderators at the moment backstab their peers and kill the authority of the Red Eclipse team this way. Also, it's sad that the old "we trust our moderators" rule doesn't work due to this misbehavior by some moderators either. I want to remind everyone that especially this should not happen, be it in public or not, but especially the former should be avoided.

**Comment by acerspyro** [comment]
I think you're gonna have to define "bad behavior" in legal terms.

**Comment by TheAssassin** [comment]
Suggestions welcome.

**Comment by qreeves** [comment]
Updated to 2nd draft.

**Comment by zaquest** [comment]
> If one moderator disagrees with the actions taken by another moderator it is advised that they discuss the situation before taking any further action.

I suggest adding an item about reaching consensus between moderators that are currently in-game before taking any actions. By voting and getting >50% of moderators' voices. Current version allows one moderator to take actions that can't be immediately reverted even if majority of moderators online disagree with the decision.

**Comment by TheAssassin** [comment]
That is covered by the paragraph about moderator misbehavior. If it is clearly not justified, the action can be undone. The section you mention is there to make sure that a moderator's actions aren't undone by non involved ones, as they now have to check with the person who issued it.

I wouldn't put in such a restrictive threshold, imagine a server with 10 moderators, one has to contact them all before taking action? That costs too much time, whispering that much, and is pretty annoying overhead, for simple stuff like a mute or quarantine, you shouldn't need more time for discussion than the measure will last for. Something like "moderators are encouraged to contact a reasonable amount of peers if there are some on the same server before taking further action" will suffice, as I think if more than one (or two or more, if there's a lot of them) moderators reach a consensus it is really time to take action.

**Comment by Jacinta-cinta** [comment]
No voting required, moderators have been given extra power for a reason. If they are found to abuse it, their rights will be stripped. As Quin wrote, if there is a disagreement it needs to be discussed. Instead of everyone acting like a spoilt child, work together and if there is an issue, bring it up with another moderator or on the forums. It is really simple!

**Comment by TheAssassin** [comment]
That's what I tried to explain to people yesterday. If you can't trust your moderators, more control and more obstacles before actions do not solve the problems. Ask yourself if they may continue to be privileged. The moderators usually discuss it anyway with a peer, as written above, so we could make it an explicit rule, too. Same goes for server owners, if you don't trust them any more, get rid of them. More surveillance doesn't help, as most suggestions may sound good, but can be proven to be weak easily. Also many of them involve serious issues regarding privacy and could be seen as illegal depending on the server location.

Apart from that, I'd like to see an explicit paragraph that states that a user may not expect any explanation on the server, because that invites for longer discussions and is even more disruptive. The consensus is, if you feel mistreated, go to the forums, there is thus no need to discuss in-game. I think personal offenses, even if they are allowed in general (I don't intend to change that) should allow for a mute after a legitimate action, to show that rules that are enforced are not to be discussed ingame.

Also, I want to add a paragraph on (intentionally) depicting situations wrongly afterwards. If someone complains, they must be honest and depict the whole situation. I do not say they must be objective, but leaving out information or taking single sentences out of context, as done very often, leads to misinterpretions easily. These complaints should not be attempted to be corrected if this is the case by adding relevant information or even by correcting sayings, but rejected immediately, to show that cheap tries to blackmail someone are not tolerated.

**Comment by zaquest** [comment]
The paragraph about depicting situations wrongly afterwords won't work well. First, if a person was banned he might not be able to get any proofs at all, since he can't join the server, thus all he can do is just file a bare complaint. Second, he can't know all the episodes that affected moderator's decision and the paragraph would mean that omitting some of them will make the complaint non-viable. Moreover, if one part of the situation happened just a day ago (or in some cases earlier the same day) it might be impossible to retrieve demos to justify oneself. Besides I think it should be moderator's job to explain his actions if they cause contradiction.

**Comment by TheAssassin** [comment]
The paragraph is about blackmailing, like, if someone posted a complaint citing something that hasn't been said in this order, omitting context to change the meaning, shortening until they sound differently etc. Also moderators *should* be able to justify, but not *have* to. On request, sure, that's really obvious. But there are situations in which this is not necessarily possible or worsens the situation more. Forcing people to explain every single step they do is like asking someone to explain every single shot, it doesn't add anything (the player knows what has been done why, if not, go file a complaint) but disadvantages the moderator greatly,  and disrupts the whole game a lot more than necessary. Complex situations can't be depicted ingame anyway. They might even include information that is not intended for public knowledge (no, not any private issues or disputes, read again if you interpreted it like that)

Furthermore, if there is no evidence available for a complaint, you cannot verify it anyway, and in these cases (word of moderator) > complaint is the general rule. There is no way to solve that, and we did not need that in the past. Please try to not overengineer nonexistent problems, everyone.

**Comment by qreeves** [comment]
4th draft. I think I've covered everything here so far as much as I'd care to.

**Comment by IceflowRE** [comment]
[Line 85-86](https://github.com/red-eclipse/base/pull/703/files#diff-2e07dd34b69681fe15193412d1cb6639R85)

> This includes sending too many messages via in-game communication over a short amount of time.

In my eyes this will forbid a normal communication, where you talk much.

**Comment by JeffBobbo** [comment]
@IceflowRE This means more like flood control, like 4 or 5+ messages a second (which we have an auto warning and mute for).

**Comment by IceflowRE** [comment]
I know but thats not what is written there.

**Comment by JeffBobbo** [comment]
Could this document also detail the automatic rules (flood control, tk, etc) we have in place, @qreeves?

**Comment by TheAssassin** [comment]
We should descriptions for the player roles including the icons (at least on the wiki), and describe their state in the community.

----

As mentioned on the forums, we wanted to liberate the rules about modifications. The following suggestion is a first step towards that goal.

```
 - * If the source code of the server is modified substantially you must contact
 -  the Red Eclipse Team to check that the changes are permitted.
 +  the Red Eclipse Team to check that the changes are permitted by creating a
 +  a topic in the "General Discussion" section of the official Red Eclipse forums
 +  located at https://redeclipse.net/forum
```

I suggest to remove the source code restriction. The actual modification to the source code doesn't necessarily have any impact on the gameplay. We could keep the idea by having modifications to the gameplay itself checked with the team. This way, someone might e.g. implement something useful, e.g. like a Lua or Python interface, for a potential modding community which is something that keeps other OSS games alive and could get us a larger player base, and test the server in public. Maybe there's additional moderation features available, or other non-gameplay features, they could be tested in public. Or, someone starts to implement an alternative server (maybe in another programming language), as long as it works like the official server, there is no real reason to put in a restriction on the master list.

---

```
 +  - It is important to note that moderators are considered to be representatives
 +    of the Red Eclipse community and should be treated with respect when
 +    performing their official duties.
```

Maybe we should add "Failure to adhere to this rule, especially after having been affected by a disciplinary measure, is not allowed, and might lead to a mute [etc.]." Otherwise, it's rather difficult to show the affected user that they did something wrong. Sure, if someone comes back after a kick, he can be banned easily, that's covered elsewhere. But measures like a quarantine can't be effective if the quarantined player starts to argue while being quarantined. A mute is probably effective then, as it is the next logical step. Though, this is not documented. Such a rule does not affect the reports mentioned elsewhere, their quality might even get better because it's easier for a moderator to justify there instead of being dragged into a discussion right after the measure. It also protects a moderator from behaving unprofessional.

---

```
+* If a player or moderator believes that the behaviour of any moderator is not
 +  supported by this document, it is advised that they lodge an official
 +  complaint in the "General Discussion" section of the official Red Eclipse
 +  forums located at https://redeclipse.net/forum
 +  - Arguing or being otherwise antagonistic about official moderator decisions
 +    via in-game communication is not allowed.
 +  - Attempting to mislead or otherwise deceive in a complaint is grounds for
 +    disciplinary action and may result in an immediate closing of the complaint.
 +  - Lodging a complaint does not guarantee that action will be taken.
 +  - The Red Eclipse Team is not obliged to disclose to the complainant what
 +    actions have or have not been taken.
```

Can we add "Furthermore, complaints which cannot be verified will be closed immediately, too.". I'd furthermore add this: "It is the complainant's duty to provide comprehensible information, such as demos or chat logs. If the complainant is not able to provide evidence, they must understand that their complaint cannot be effective. General complaints without evidence are not allowed, and might in extreme cases lead to further disciplinary actions." It is not the moderators' duty to bring up evidence for a complaint, often this isn't even possible. To avoid shitstorms and pointless internal discussions about a non-verifiable complaint, there should be the possibility to reject them. This should't be used as a cheap excuse to get rid of complaints though, just create the possibility to do this.

---

I'd also like to add a paragraph about measures that have been taken accidentally due to a problem when trying to communicate with a user, e.g. when a user does not react to a warning, and is thus kicked or quarantined or muted. A moderator cannot really be accountable for such a misunderstanding. The primary language is English, and since we got quite a few players who aren't very good at it or don't even attempt to understand statements in this language, moderators do have to take further action. This does not mean that e.g. Spanish speaking moderators may not moderate the game in Spanish as well, same goes for any other language. I couldn't get up with a proper wording yet, contributions welcome.

---

For my last suggestion, I'm getting into a slight moral conflict. I am a great defendant of the human rights, especially the freedom of speech and related fundamental rights. Though, I see that moderation of social media (to which a game and its ecosystem can belong) can be and usually is necessary. The problem is to create a set of rules that are effective though. Big companies (like a huge social network with a blue logo) have lots of problems with this, and I guess we are likely to run into the same issues.

The idea is that we have had, as @acerspyro called it, "toxic" players that have tried to breed discord within the community, between players, moderators, doesn't really matter. We have had such a case with one of the people currently on the global mute list. He has received a global mute and noone really questioned this. Though, there is no real rule about that in the guidelines yet. To maintain a good atmosphere in the community, the team must be able to remove such "trolls" from it. For this to work in a non-disputable way, we might have to find a way to properly classify these players. Many people get triggered by such things, and without carefully validating everything that is said, it is difficult to converse properly and with a positive attitude. Situations can escalate easily.

I guess anyone can tell a troll if they see one, though recently, we have this "this is not explicitly regulated" or "there is no such rule" attitude among some individual players who do not accept the interpretation of the moderators about single rules, and some moderators failed to check for their peer's interpretation to be able to compare the different interpretations. We either need to disallow people to argue about the way team members interprets rules and keep explaining these for single rules if required, or we need to add more regulation.

The problem with the latter way is that it might conflict with the freedom of speech quite easily if the team is not very, very careful about it.

Thus, I highly recommend the first approach. As moderators have been put to their roles based on the operation team trusting them enough, this trust should extend to the ability of being able to interpret the rules correctly.

My approach to this is to add a weaker rule about it, just to have an explicit paragraph to refer to, about toxic behavior:

"The Red Eclipse team does not tolerate toxic behavior by any community member. Moderators are encouraged to step in and stop attempts to do so."

For the interpretation rule, I'd say that this will work: "Users may not argue about moderators' interpretations of rules for enforcement. If they feel there was a mistake, they should post about it on the forums. The team will then review both sides, and attempt to find a solution."

**Comment by IceflowRE** [comment]
Removing the source code restriction is a good idea. In my opinion it should be removed completly, even the changed gameplay aspect. In addition we could add badges to the server menu, which mark them as changed gameplay.
This would be the best for all.

**Comment by TheAssassin** [comment]
> In my opinion it should be removed completly, even the changed gameplay aspect.

I understand the paragraph as a chance for us to get an idea what people are trying to do with the game. But I can see that it kind of implies that we aren't open for such things. Maybe we should rather reword it to make it sound differently.

> In addition we could add badges to the server menu, which mark them as changed gameplay.

Mind to explain your idea a bit further? I don't see how we could technically do that, with the official server code we've got the modification ratio, but who says that might be implemented in a third party server?

Both the rule you refer to and the one requiring a note in the `serverdesc`/`servermotd` ensure that players are able to see when something has changed. As stated above, I'd rather reword it but I like keeping the idea.

**Comment by acerspyro** [comment]
> Mind to explain your idea a bit further? I don't see how we could technically do that, with the official server code we've got the modification ratio, but who says that might be implemented in a third party server?

Just specify in the guidelines that this part of the code has to be left in, and that if an administrator finds out your server is modified but doesn't show the badge, your server risks being blacklisted from the master server list.

**Comment by TheAssassin** [comment]
That opens the door for quite some abuse of this rule, then we can just leave it out completely, and save us from the hassle of banning servers who do not adhere to rules. If people check their stuff with the team before getting them on the master list avoids most of that.

>  this part of the code has to be left in

Especially this part is impossible to realize with servers implemented completely independently, or written in a language like Go (which is quite popular for these kinds of things).

It is essentially the same as with blacklisting vs. whitelisting. A good article about it is https://www.schneier.com/blog/archives/2011/01/whitelisting_vs.html. There's many parallels, and the whitelisting approach is always the better choice.

**Comment by acerspyro** [comment]
> Especially this part is impossible to realize with servers implemented completely independently, or written in a language like Go (which is quite popular for these kinds of things).

My fault for assuming that all servers would be based off the current codebase.

**Comment by TheAssassin** [comment]
*Offtopic:* I already started an experiment by implementing the essential parts in Python (using PyPy3) and asyncio. It worked great as far as I developed it (implemented anonymous connects). The only challenge is the lack of documentation and the state of the current codebase being hard to read and understand.

**Comment by TheAssassin** [comment]
Half of the conversation wasn't regarded any more, 11 comments. @qreeves I'd like to know the reason.

**Comment by qreeves** [comment]
I'm not interested in arguing about the code modification clauses for the umpteenth time. It's been discussed to death and I'm not going to change my position on the matter. It isn't hard to ask us to look at any submissions.

**Comment by TheAssassin** [comment]
> I'm not interested in arguing about the code modification clauses for the umpteenth time. 

Oh come on. That is just a little fraction of the whole conversation.
